ci-operator/config/openshift-eng/agent-qe-infra/tools/generate_bm_cron.py

```
@@ -0,0 +1,229 @@
+    # Global tracking registers using composite key: (day_of_month, slot_name)
+    global_capacity = defaultdict(int)
+    job_weight_map = {}
+    valid_paths = [Path(d) for d in yaml_dirs if Path(d).exists()]
+
+    # Pre-scan files to identify valid jobs and aggregate weights
+    for dir_path in valid_paths:
+        for yaml_file in dir_path.rglob("*.yaml"):
+            try:
+                content = yaml_file.read_text()
+            except Exception:
+                continue
```

Ruff (0.15.20)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)